src/composed/any.rs

```diff
@@ -0,0 +1,68 @@
+use std::io::{BufRead, Read};
+
+use buffer_redux::BufReader;
+
+use crate::{
+    armor::{self, BlockType, Dearmor},
+    cleartext::CleartextSignedMessage,
+    errors::Result,
+    Deserializable, Message, SignedPublicKey, SignedSecretKey, StandaloneSignature,
+};
+
+/// A flexible representation of what can be represented in an armor file.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum Any {
+    Cleartext(CleartextSignedMessage),
+    PublicKey(SignedPublicKey),
+    SecretKey(SignedSecretKey),
+    Message(Message),
+    Signature(StandaloneSignature),
+}
+
+impl Any {
```

Comment:
This name clashes with `std::any::Any`. I wonder if the name is worth changing.

I fear the RFC doesn't offer terminology for "any kind of PGP thing". The most generic objects it names are "Messages" and "Keys" (neither of which includes cleartext messages).

Attempting to brainstorm alternative names: `(Open)PgpData`, `(Open)PgpContent`.

I currently have no idea that I'm excited about.

(I don't think this kind of generic reader needs to be limited to armored data, so I would tend to not use "armor" in the naming)

Reply:
I think I am going to stick with `Any` for now, and wait for better suggestions :)